src/api/services/translations_cache.py

```diff
@@ -0,0 +1,128 @@
+        """
+        translation, err = self.store.get_translation(context, args)
+        return translation if translation else None, err
+
+    def get_target_languages (self, context: Context, target_language_code) -> Tuple[ Union[list, None], None ]:
+        """
+        This function returns the target languages for translation except the source (default) language.
+        It calls the SupportedLanguageStore.list_supported_languages method to get all supported languages and then filters out the source language.
+
+        Args:
+        - context: Context
+        - target_language_code: str
+
+        Returns:
+        - target_languages: list
+        """
+        supported_languages, err = SupportedLanguageStore().list_supported_languages(context, { })
+        target_languages = []
+
+        if err:
+            return None, err
+
+        for language in supported_languages:
+            if language.code != DEFAULT_SOURCE_LANGUAGE_CODE:
+                target_languages.append(language)
+
+        return target_languages, err
+
+    def create_list_of_all_records_to_translate (self, models):
+        """
+        This function creates a list of all records to translate
+
+        Args:
+        - models: list
+
+        Returns:
+        - records: list
+        """
+
+        all_records = [ ]
+        for model in models:
+
+            translation_meta = getattr(model, "TranslationMeta", None)
+            if not translation_meta:
+                continue
+
+            translatable_fields = getattr(translation_meta, "fields_to_translate", None)
+            if len(translatable_fields) > 0:
+                all_records.extend([ model_to_dict(r, fields = translatable_fields) for r in model.objects.all() ])
+
+        return all_records
+
+    def translate_all_models (self, language_code: str) -> Tuple[ dict or None, None ]:
+        try:
+            # Go through all the records of all models and return a list of all records (the translatable fields for each) [{:dict}] to translate
+            all_records= self.create_list_of_all_records_to_translate(apps.get_models())
+
+            source_language = to_third_party_lang_code(DEFAULT_SOURCE_LANGUAGE_CODE)
+            target_language = to_third_party_lang_code(language_code)
+
+            json_translator = JsonTranslator(all_records)
+            _, translations, hashes = json_translator.translate(source_language,target_language)
+
+            for i in range(0, len(translations)):
+                translation, err = self.create_translation( {
+                    "hash": hashes[i],
+                    "source_language": DEFAULT_SOURCE_LANGUAGE_CODE,
+                    "target_language": language_code,
+                    "translated_text": translations[i]
+                })
+
+                if err:
+                    log.error(f"Error caching translation for {hashes[i]}: {err}")
+                    return None, err
```

Comment:
i like the idea of this.  What do you think about putting a TODO comment here to eventually make this process async.  For instance: we can put all the things to translate in an SQS queue.  then we have something that polls the queue and processes anything that is in that queue by performing this action you have here.  If the creation of the translation fails, we will add it to a dead letter queue so we can track all the insertions that didnt work and can alert on it.  And runs some things manually if need be.  For now this is ok but let's leave a comment to do that in the near future.  I am also open to ideas here.

That said if an error occurred in calling ` self.create_translation` lets not return an error.  Just because one failed does not mean we stop everything.  Maybe the next one will succeed.  So at the very least we should rather be collecting the errors and maybe log some message that X errors occurred.  This is where having that queue would be helpful in the future because if at first it does not succeed, the system will auto retry at some other point.

Reply:
I've rewritten this function to be significantly more robust than the previous one, added retry logic as well as changed the processing by using a deque queue instead of the list.
A pseudo dead letter queue has been added currently for error reporting and also to make it easy to leverage SQS when we're ready.

Reply:
SIDE NOTE @Opoku-Agyemang :
I rebased main into this to get the models in. Now the files have increased in number but focus on:
`src/api/services/translations_cache.py`

I made the changes requested in `src/api/services/text_hash.py` and `src/api/store/text_hash.py` but also marked them as deprecated with respect to our last discussion